src/io/buf_writer.rs

```diff
@@ -301,80 +293,3 @@ impl<W: AsyncWrite + AsyncSeek> AsyncSeek for BufWriter<W> {
         self.inner().poll_seek(cx, pos)
     }
 }
-
-mod tests {
-    #![allow(unused_imports)]
-
-    use super::BufWriter;
-    use crate::io::{self, SeekFrom};
-    use crate::prelude::*;
-    use crate::task;
-
-    #[test]
-    fn test_buffered_writer() {
-        task::block_on(async {
-            let inner = Vec::new();
-            let mut writer = BufWriter::with_capacity(2, inner);
-
-            writer.write(&[0, 1]).await.unwrap();
-            assert_eq!(writer.buffer(), []);
-            assert_eq!(*writer.get_ref(), [0, 1]);
-
-            writer.write(&[2]).await.unwrap();
-            assert_eq!(writer.buffer(), [2]);
-            assert_eq!(*writer.get_ref(), [0, 1]);
-
-            writer.write(&[3]).await.unwrap();
-            assert_eq!(writer.buffer(), [2, 3]);
-            assert_eq!(*writer.get_ref(), [0, 1]);
-
-            writer.flush().await.unwrap();
-            assert_eq!(writer.buffer(), []);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3]);
-
-            writer.write(&[4]).await.unwrap();
-            writer.write(&[5]).await.unwrap();
-            assert_eq!(writer.buffer(), [4, 5]);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3]);
-
-            writer.write(&[6]).await.unwrap();
-            assert_eq!(writer.buffer(), [6]);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3, 4, 5]);
-
-            writer.write(&[7, 8]).await.unwrap();
-            assert_eq!(writer.buffer(), []);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3, 4, 5, 6, 7, 8]);
-
-            writer.write(&[9, 10, 11]).await.unwrap();
-            assert_eq!(writer.buffer(), []);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
-
-            writer.flush().await.unwrap();
-            assert_eq!(writer.buffer(), []);
-            assert_eq!(*writer.get_ref(), [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11]);
-        })
-    }
-
-    #[test]
-    fn test_buffered_writer_inner_into_inner_does_not_flush() {
-        task::block_on(async {
-            let mut w = BufWriter::with_capacity(3, Vec::new());
-            w.write(&[0, 1]).await.unwrap();
-            assert_eq!(*w.get_ref(), []);
-            let w = w.into_inner();
-            assert_eq!(w, []);
-        })
-    }
-
-    #[test]
-    fn test_buffered_writer_seek() {
-        task::block_on(async {
-            let mut w = BufWriter::with_capacity(3, io::Cursor::new(Vec::new()));
-            w.write_all(&[0, 1, 2, 3, 4, 5]).await.unwrap();
-            w.write_all(&[6, 7]).await.unwrap();
-            assert_eq!(w.seek(SeekFrom::Current(0)).await.ok(), Some(8));
-            assert_eq!(&w.get_ref().get_ref()[..], &[0, 1, 2, 3, 4, 5, 6, 7][..]);
-            assert_eq!(w.seek(SeekFrom::Start(2)).await.ok(), Some(2));
-        })
-    }
-}
```

Comment:
I kind of like these inline tests though; we have them for some of the `sync` modules as well.

Reply:
(don't feel strongly about this at all; but figured I'd still mention it :blush:)

Reply:
IIRC, this removes the need to recompile crate every time you change unit tests.

Reply:
Only for `Barrier`, right? Tests for `Mutex` and `RwLock` are in the `tests` directory. I don't have a strong preference either, but would like to be consistent about it :)

My general rule of thumb is that if tests are only poking the public API, they should go into the `tests` directory. If they are checking internal (private) state of the thing that is being tested, it should go inside the module where it's implemented.

Reply:
Seems fair enough!